homeassistant/components/sanix/sensor.py

```diff
@@ -0,0 +1,125 @@
+"""Platform for Sanix integration."""
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from datetime import date, datetime
+
+from sanix.const import (
+    ATTR_API_BATTERY,
+    ATTR_API_DEVICE_NO,
+    ATTR_API_DISTANCE,
+    ATTR_API_FILL_PERC,
+    ATTR_API_SERVICE_DATE,
+    ATTR_API_SSID,
+)
+from sanix.models import Measurement
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import PERCENTAGE, UnitOfLength
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceEntryType, DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN, MANUFACTURER
+from .coordinator import SanixCoordinator
+
+
+@dataclass(frozen=True, kw_only=True)
```

Comment:
The reason tests failed was because `kw_only=True` was missing

Reply:
That was quick ;D thanks a lot